### ISSUE TITLE
Shape Mismatch in Synthetic Features and Labels 

### ISSUE THREAD

**Issue by chanztuying** [ChandlerBang/GCond#12]
I've encountered a shape mismatch issue between synthetic features (feat_syn) and labels (labels_syn) when condensing the ogbn-arxiv dataset using a reduction rate of 0.0005. The method ensures each class is represented by at least one label, but the calculation feat_train * reduction_rate can lead to a discrepancy in the number of features and labels.

Probably consider using `n = max(int(len(data.idx_train) * args.reduction_rate), labels_syn.shape[0])` to calculate the number of synthetic features?

**Comment by rockcor** [comment]
You said it. We improved it and you can replace the start of `condensation/gcond_agent_transduct.py` with below.
```python
  def __init__(self, data, args, device='cuda', **kwargs):
        self.data = data
        self.args = args
        self.device = device

        # n = data.nclass * args.nsamples

        self.data.labels_syn = self.generate_labels_syn(data)
        n = self.data.labels_syn.shape[0]
        self.nnodes_syn = n
        d = data.feat_train.shape[1]
        print(f'target reduced size:{int(data.feat_train.shape[0] * args.reduction_rate)}')
        print(f'actual reduced size:{n}')

        # from collections import Counter; print(Counter(data.labels_train))

        self.feat_syn = nn.Parameter(torch.FloatTensor(n, d).to(device))
        self.pge = PGE(nfeat=d, nnodes=n, device=device, args=args).to(device)

        self.reset_parameters()
        self.optimizer_feat = torch.optim.Adam([self.feat_syn], lr=args.lr_feat)
        self.optimizer_pge = torch.optim.Adam(self.pge.parameters(), lr=args.lr_adj)
        print('adj_syn:', (n, n), 'feat_syn:', self.feat_syn.shape)

    def reset_parameters(self):
        self.feat_syn.data.copy_(torch.randn(self.feat_syn.size()))

    def generate_labels_syn(self, data):
        if not hasattr(data, 'labels_train'):
            data.labels_train = data.labels[data.idx_train]
        counter = Counter(data.labels_train)
        num_class_dict = {}
        # n = len(data.labels_train)

        sorted_counter = sorted(counter.items(), key=lambda x: x[1])
        sum_ = 0
        labels_syn = []
        self.syn_class_indices = {}
        for ix, (c, num) in enumerate(sorted_counter):
            num_class_dict[c] = max(int(num * self.args.reduction_rate), 1)
            sum_ += num_class_dict[c]
            self.syn_class_indices[c] = [len(labels_syn), len(labels_syn) + num_class_dict[c]]
            labels_syn += [c] * num_class_dict[c]

        self.num_class_dict = num_class_dict
        return np.array(labels_syn)
```
